delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/AclsRoutes.scala

```diff
@@ -0,0 +1,275 @@
+
+import akka.http.javadsl.server.Rejections.validationRejection
+import akka.http.scaladsl.model.StatusCodes._
+import akka.http.scaladsl.model.Uri.Path
+import akka.http.scaladsl.model.Uri.Path._
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.{Directive1, MalformedQueryParamRejection, Route}
+import cats.implicits.{toBifunctorOps, toFunctorOps}
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.contexts
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.JsonLdEncoder
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.{ContextValue, RemoteContextResolution}
+import ch.epfl.bluebrain.nexus.delta.rdf.utils.JsonKeyOrdering
+import ch.epfl.bluebrain.nexus.delta.routes.AclsRoutes.PatchAcl._
+import ch.epfl.bluebrain.nexus.delta.routes.AclsRoutes._
+import ch.epfl.bluebrain.nexus.delta.routes.marshalling.{CirceUnmarshalling, QueryParamsUnmarshalling}
+import ch.epfl.bluebrain.nexus.delta.sdk.directives.AuthDirectives
+import ch.epfl.bluebrain.nexus.delta.sdk.model.ResourceF._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.acls.AclAddress.{Organization, Project}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.acls.AclAddressFilter.{AnyOrganization, AnyOrganizationAnyProject, AnyProject}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.acls.{Acl, AclAddress, AclAddressFilter, AclRejection}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity
+import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.Permission
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchResults
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchResults.{encodeResults, SearchEncoder}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, Label, ResourceF}
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.delta.sdk.{AclResource, Acls, Identities, Lens}
+import io.circe.generic.semiauto.deriveDecoder
+import io.circe.syntax.EncoderOps
+import io.circe._
+import monix.execution.Scheduler
+
+class AclsRoutes(identities: Identities, acls: Acls)(implicit
+    baseUri: BaseUri,
+    s: Scheduler,
+    cr: RemoteContextResolution,
+    ordering: JsonKeyOrdering
+) extends AuthDirectives(identities, acls)
+    with DeltaDirectives
+    with CirceUnmarshalling
+    with QueryParamsUnmarshalling {
+
+  import baseUri._
+
+  private val aclsIri = endpoint.toIri / "acls"
+
+  private val any = "*"
+
+  private val simultaneousRevAndAncestorsRejection =
+    MalformedQueryParamRejection("rev", "rev and ancestors query parameters cannot be present simultaneously")
+
+  implicit val iriLens: Lens[AclAddress, Iri] = {
+    case AclAddress.Root    => aclsIri
+    case Organization(org)  => aclsIri / org.value
+    case Project(org, proj) => aclsIri / org.value / proj.value
+  }
+
+  implicit val aclContext: ContextValue = Acl.context
+
+  implicit val aclResponseEncoder: Encoder.AsObject[AclResource] = Encoder.AsObject
```

Comment:
Some of this implicits might not be necessary. But we'll be able to refactor this once I simplify the `ResourceF` (which also yields some changes on the Acl structure)

Reply:
I think we still need them, because ACLs are encoded a bit differently than other resources:
-  they have `_path` as part of the metadata
- they don't have `_deprecated`

Reply:
`AclResource` is defined as 
```scala
type AclResource = ResourceF[AclAddress, Acl]
```
and `Acl` doesn't have the `AclAddress` to make `_path` from the resource value

Reply:
Don't worry about this point for now. With the changes I've introduced in `ResourceF[A]` `Acl` will actually have the `AclAddress`. But you can keep as is now.